### PR TITLE
change ut to c99 and turn off cppunittest by default

### LINKED ISSUE #1923: azure-iot-sdk-c_x86 failed to build after upgrading CMake to 3.19.4 or later
Hi All,

After we upgrading cmake to 3.19.6, azure-iot-sdk-c x86 failed to build on MSVC. Verified that we can get a clean build when the cmake version is lower than 3.19.4.

I compare the build command line, and found "CPP_UNITTEST" "USE_CTEST" "/TP" "/TC" does not seem to match correctly. This  may be related to CMake Policy CMP0077(cmake --help-policy CMP0077).

**Build steps:**

1. open VS 2019 x86 native tools command
2. git clone --recursive http://www.github.com/azure/azure-iot-sdk-c  azure_iot_sdk_c
3. mkdir build_x86 && pushd build_x86
4. cmake -G "Visual Studio 16 2019" -A Win32 -DCMAKE_SYSTEM_VERSION=10.0.18362.0 -Drun_unittests:BOOL=ON .. >cmake.log
5. msbuild /m /p:Platform=Win32 /p:Configuration=Release azure_iot_sdks.sln /t:Rebuild > build.log

**Build log:** 
[build_cmake_v3.17.0-passed.log](https://github.com/Azure/azure-iot-sdk-c/files/6248306/build_cmake_v3.17.0-passed.log)
[build_cmake_v3.19.6-failed.log](https://github.com/Azure/azure-iot-sdk-c/files/6248305/build_cmake_v3.19.6-failed.log)

**Error message:**
F:\gitP\azure\azure-iot-sdk-c\build_x86>cmake --version
**cmake version 3.17.0**
F:\gitP\azure\azure-iot-sdk-c\build_x86>CL.exe /c /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-macro-utils-c\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\umock-c\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\pal\windows" /I"F:\gitP\azure\azure-iot-sdk-c\serializer\..\deps\parson" /I"F:\gitP\azure\azure-iot-sdk-c\serializer\inc" /I"F:\gitP\azure\azure-iot-sdk-c\iothub_client\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\testtools\micromock\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-c-testrunnerswitcher\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-ctest\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\src" /I"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\UnitTest\include" /nologo /W4 /WX /diagnostics:column /O2 /Ob2 /Oy- /D CPPUNITTEST_SYMBOL=some_symbol_for_cppunittest_1 /D WIN32 /D _WINDOWS /D NDEBUG **/D CPP_UNITTEST** /D _CRT_SECURE_NO_WARNINGS /D ARCHITECTURE_x86=1 /D GB_MEASURE_MEMORY_FOR_THIS /D GB_DEBUG_ALLOC /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"serializer_dt_ut_testsonly_lib.dir\Release\\" /Fd"serializer_dt_ut_testsonly_lib.dir\Release\serializer_dt_ut_testsonly_lib.pdb" /Gd **/TP** /wd4232 /analyze- /errorReport:queue "F:\gitP\azure\azure-iot-sdk-c\serializer\tests\serializer_dt_ut\serializer_dt_ut.c" -I"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\VS\UnitTest\include"
serializer_dt_ut.c
F:\gitP\azure\azure-iot-sdk-c\build_x86>

--------------------------------------------------------------------------------------------------------------

F:\gitP\azure\azure-iot-sdk-c\build_x86>cmake --version
**cmake version 3.19.6**
F:\gitP\azure\azure-iot-sdk-c\build_x86>CL.exe /c /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-macro-utils-c\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\umock-c\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\pal\windows" /I"F:\gitP\azure\azure-iot-sdk-c\serializer\..\deps\parson" /I"F:\gitP\azure\azure-iot-sdk-c\serializer\inc" /I"F:\gitP\azure\azure-iot-sdk-c\iothub_client\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\testtools\micromock\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-c-testrunnerswitcher\inc" /I"F:\gitP\azure\azure-iot-sdk-c\deps\azure-ctest\inc" /I"F:\gitP\azure\azure-iot-sdk-c\c-utility\src" /I"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\UnitTest\include" /nologo /W4 /WX /diagnostics:column /O2 /Ob2 /Oy- /D CPPUNITTEST_SYMBOL=some_symbol_for_cppunittest_1 /D WIN32 /D _WINDOWS /D NDEBUG **/D CPP_UNITTEST** /D _CRT_SECURE_NO_WARNINGS /D ARCHITECTURE_x86=1 /D GB_MEASURE_MEMORY_FOR_THIS /D GB_DEBUG_ALLOC /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"serializer_dt_ut_testsonly_lib.dir\Release\\" /Fd"serializer_dt_ut_testsonly_lib.dir\Release\serializer_dt_ut_testsonly_lib.pdb" /Gd **/TC** /wd4232 /analyze- /errorReport:queue "F:\gitP\azure\azure-iot-sdk-c\serializer\tests\serializer_dt_ut\serializer_dt_ut.c" -I"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\VS\UnitTest\include"
serializer_dt_ut.c
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(23,62): error C2061: syntax error: identifier 'noexcept'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(23,62): error C2059: syntax error: ';'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(23,90): error C2449: found '{' at file scope (missing function header?)
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(25,1): error C2059: syntax error: '}'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(31,72): error C2061: syntax error: identifier 'noexcept'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include\cstdlib(31,72): error C2059: syntax error: ';'

### PR DESCRIPTION
**Pull request by danewalton**
Fixes #1923

Okay two issues here. First, as described in the linked GH issue, newer versions of CMake are swallowing the `/TP` flag when trying to compile `.c` source files as CPP. Here I've turned off the `use_cppunittest` by default so users don't get that on automatically. Unit tests still build as `.exe`'s and they can still run tests if they are on newer versions of CMake. Until we get a response from CMake about the long term solution to this regression we will turn on `use_cppunittest` at the gate where we have older versions of CMake that don't break the build.

Second issue is that new versions of CMake are adding the C11 flag to `.vcxproj` files where before they weren't. This is breaking our build and as we are C99 compliant not C11, I have changed these UT's to be C99. Below is the change to the output `.vcxproj` files.

**Before changing to C99:**

*CMake 3.15*

before315

*CMake 3.20*

before320

After changing to C99:

*CMake 3.15*

after315

*CMake 3.20* 

after320

### CODE REVIEW

File: iothub_client/tests/iothubtransport_ut/iothubtransport_ut.c [hunk 1]
```diff
@@ -31,9 +31,6 @@ static void my_gballoc_free(void* ptr)
 
 #include "iothub_transport_ll.h"
 
-#define IOTHUB_TRANSPORT_H
```
Comment (danewalton) [inline iothub_client/tests/iothubtransport_ut/iothubtransport_ut.c:34]: @jebrando long time no talk :)

Do you remember at all why this was done? At least the `_LL_H` macro would already be defined by the above include for `_ll.h`. The tests seem to pass.

File: CMakeLists.txt [hunk 2]
```diff
@@ -234,7 +234,7 @@ set(original_skip_samples ${skip_samples})
 set(run_e2e_tests OFF)
 set(run_unittests OFF)
 set(skip_samples ON)
-set(use_cppunittest ON)
+set(use_cppunittest OFF)
```
Comment (ericwol-msft) [inline CMakeLists.txt:237]: turn this off for all platforms...or only win32?

**Review by ericwol-msft — APPROVED** [review]

**Comment by ericwol-msft** [comment]
make sure we have a card to track CMAKE fixing this issue.

File: iothub_client/tests/iothubtransport_ut/iothubtransport_ut.c (hunk 1 above)
Comment (danewalton) [reply iothub_client/tests/iothubtransport_ut/iothubtransport_ut.c:34]: discussed offline

File: CMakeLists.txt (hunk 2 above)
Comment (danewalton) [reply CMakeLists.txt:237]: this switch only affects WIN32 in the tests.
https://github.com/Azure/azure-c-shared-utility/blob/341c40b8ce15d2372d9407ef537b0abe45c452d4/configs/azure_c_shared_utilityFunctions.cmake#L232-L240

**Comment by danewalton** [comment]
> make sure we have a card to track CMAKE fixing this issue.

Added an issue [here](https://gitlab.kitware.com/cmake/cmake/-/issues/22111)

**Pull request merged**